subsys/storage/flash_map/flash_map_shell.c

```diff
@@ -12,6 +12,7 @@
 #include <ctype.h>
 #include <zephyr/storage/flash_map.h>
 #include <zephyr/logging/log.h>
+#include <zephyr/device.h>
```

Comment:
Why is this necessary?

Reply:
The struct for device is missing which generates an incomplete type compiler error.
```
In file included from /Users/stevenslupsky/Documents/source/zephyrproject/zephyr/subsys/storage/flash_map/flash_map_shell.c:7:
/Users/stevenslupsky/Documents/source/zephyrproject/zephyr/subsys/storage/flash_map/flash_map_shell.c: In function 'fa_cb':
/Users/stevenslupsky/Documents/source/zephyrproject/zephyr/subsys/storage/flash_map/flash_map_shell.c:27:46: error: dereferencing pointer to incomplete type 'const struct device'
       fa->fa_id, fa->fa_device_id, fa->fa_dev->name,
                                              ^~
```

Reply:
I recall the error was introduced with PR #47193 to remove device_get_binding.